### PR TITLE
Create views for user and post views

### PR DESCRIPTION
**Pull request by GutemaG**
### Project requirements
- Implement the 
Screenshot from 2023-04-05 21-35-25
design from the sneak peek the  wireframes. (:heavy_check_mark: )

    - Styling is nice to have but focus on making sure everything functions first. (Note: Ignore login logic, you would be using Devise in coming milestone)
-

- Create views for show and index for each post and user controller

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1,2 +1,15 @@
...
-<h2>Posts</h2>
+<div class="container p-2">
+  <div class="row flex flex-column">
+    <%= render "shared/user_header", user: @user%>
+    <div class="bio border p-3 w-50">
+      <h4>Bio</h4>
+      <p> <%= @user.bio%> </p>
+    </div>
+    <div class="w-50 border p-3 mt-1">
+      <% @user.posts.each do |post| %>
```
Comment (julie-ify) [inline app/views/users/show.html.erb]: - [ ] The posts that should be rendered in this user show page are the recent posts of the user. Kindly refactor this view to use the `recent_three_posts` in your User model.

  ```
  @user.recent_three_posts.each do |post|
    ...
  ```

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -1 +1,22 @@
...
+  <% @posts.each do |post| %>
+    <div class="w-50 border border-2 p-2">
+      <h4> <%= link_to post.title, user_post_path(post, user_id: @user.id) %> </h4>
+      <p> <%= post.text %> </p>
+      <div class="d-flex justify-content-end">
+        <span class="">Comments: <%= post.comments.count%> </span>, 
+        <span class="ml-2" > Comments: <%= post.likes.count%> </span>
+      </div>
+      <div class="border mt-1 p-2">
+        <% post.comments.each do |comment| %>
```
Comment (julie-ify) [inline app/views/posts/index.html.erb]: - [ ] The comments to be rendered on this post index page should use the recent comments method you have in the Post model, kindly do that.

File: app/views/users/index.html.erb [hunk 3]
```diff
@@ -1,8 +1,5 @@
...
-<ul>
-  <% @users.each do |user| %>
-    <li>
-      <%= link_to user.name, user_path(user) %>
-    </li>
+<div class="container p-2">
+  <%@users.each do |user| %>
+    <%= render "shared/user_header", user: user%>
   <% end %>
-</ul>
```
Comment (julie-ify) [inline app/views/users/index.html.erb]: - [ ] Good job styling the user element on the page, however, you need to center the elements on the page to follow [the wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_root_users.png). Kindly also make it larger to cover more width on the page.

  isos

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -1,2 +1,15 @@
...
+        <%= render "shared/post_comment", post: post %>
+      <% end %>
+    </div>
+    <%= link_to "See All Post", user_posts_path(@user), class:"btn btn-secondary w-50 mt-2" %>
+  </div>
```
Comment (julie-ify) [inline app/views/users/show.html.erb]: - [ ] Kindly center the elements as well to match [the wireframe](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/blog-app/images/blog_user_page.png?token=GHSAT0AAAAAACBA7J7KGYMVA3EGE46DPJMEZBN2DTQ) and make them wider.

  image

File: app/views/posts/index.html.erb [hunk 5]
```diff
@@ -1 +1,22 @@
...
+          <p>
+            <span> <%= comment.author.name %> : </span>
+            <span> <%= comment.text %> </span>
+          </p>
+        <%end%>
+      </div>
+    </div>
+  <% end %>
+  <button class="btn btn-secondary btn-sm mt-5">Pagination comming soon</button>
```
Comment (julie-ify) [inline app/views/posts/index.html.erb]: - [ ] Kindly center the elements to match [the wireframe](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/blog-app/images/blog_user_all_posts.png?token=GHSAT0AAAAAACBA7J7KYPXEYKZOH5JANANQZBN2FFQ).

  image

File: app/views/posts/show.html.erb [hunk 6]
```diff
@@ -1 +1,22 @@
...
+    <div class="border mt-1 p-2">
+      <% @post.comments.each do |comment| %>
+        <p>
+          <span> <%= comment.author.name %> : </span>
+          <span> <%= comment.text %> </span>
+        </p>
+      <%end%>
+    </div>
+  </div>
+</div>
```
Comment (julie-ify) [inline app/views/posts/show.html.erb]: - [ ] Kindly center the elements to match [the wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_single_post.png).

  image

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @GutemaG,

Good job so far!

## Highlights!
- Nice looking design ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by victorokoroji — APPROVED** [review]
Hi @GutemaG, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request is following the Git Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The commit messages are meaningful ✅ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**